### PR TITLE
wstETH pricer

### PR DESCRIPTION
**Pull request by haythem96**
# wstETH pricer

## High Level Description

This PR add `WstethPricer.sol` to return the price of wstETH token in USD scaled by 8 decimals.

### Code

- [ ] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [ ] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

**Comment by haythem96** [comment]
There is 66 files changed because of linting.

File: contracts/pricers/WstethPricer.sol [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.6.10;
+
+import {OracleInterface} from "../interfaces/OracleInterface.sol";
+import {OpynPricerInterface} from "../interfaces/OpynPricerInterface.sol";
+import {WSTETHInterface} from "../interfaces/WSTETHInterface.sol";
+import {SafeMath} from "../packages/oz/SafeMath.sol";
+
+/**
+ * @notice A Pricer contract for a Yearn yToken
```
Comment (chuddster) [inline contracts/pricers/WstethPricer.sol:10]: pricer contract for wsteth

File: contracts/pricers/WstethPricer.sol [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     */
+    function getPrice() external view override returns (uint256) {
+        uint256 underlyingPrice = oracle.getPrice(address(underlying));
+        require(underlyingPrice > 0, "WstethPricer: underlying price is 0");
+        return _underlyingPriceToWstethPrice(underlyingPrice);
+    }
+
+    /**
+     * @notice set the expiry price in the oracle
+     * @dev requires that the underlying price has been set before setting a yToken price
```
Comment (chuddster) [inline contracts/pricers/WstethPricer.sol:56]: setting wsteth price

**Review by chuddster — APPROVED** [review]

File: contracts/pricers/WstethPricer.sol [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    function _underlyingPriceToWstethPrice(uint256 _underlyingPrice) private view returns (uint256) {
+        uint256 stEthPerWsteth = wstETH.stEthPerToken();
+
+        return stEthPerWsteth.mul(_underlyingPrice).div(1e18);
+    }
+
+    function getHistoricalPrice(uint80 _roundId) external view override returns (uint256, uint256) {
+        revert("WstethPricer: Deprecated");
+    }
+}
```
Comment (CruzMolina) [inline contracts/pricers/WstethPricer.sol:80]: Can you optimize the `require` & `revert` strings like we did for the [`StakedaoEcrvPricer`](https://github.com/opynfinance/perp-vault-templates/blob/stakedao-master/contracts/utils/StakedaoEcrvPricer.sol#L12-L21)?

Here is the [diff](https://github.com/opynfinance/perp-vault-templates/pull/38/files#diff-fc97ea9e7fd83c47df786ce52cf7ecf6f0bb6219b11353942d38c67f01f57539R9).

**Review by CruzMolina — COMMENTED** [review]
Mostly good to go, one optimization to make.

File: contracts/pricers/WstethPricer.sol (hunk 3 above)
Comment (CruzMolina) [reply contracts/pricers/WstethPricer.sol]: Done

**Review by CruzMolina — APPROVED** [review]
LGTM

**Review by aparnakr — APPROVED** [review]

**Review by aleone — APPROVED** [review]

**Pull request merged**